File: Iniciante/1914.py
lim = int(input());

for i in range(lim):
  nome1, e1, nome2, e2 = input().split();
  n1, n2 = map(int, input().split());
  total = n1+n2;

  if (total%2 == 0):
    if (e1 == 'PAR'):
      print(nome1);
    else:
      print(nome2);

  else:
    if (e1 == 'IMPAR'):
      print(nome1);
    else:
      print(nome2);

File: Iniciante/1914.js
let lim = Number(lines.shift());

for (let i = 0; i < lim; i++) {
  let [nome1, e1, nome2, e2] = lines.shift().split(' ');
  let [n1, n2] = lines.shift().split(' ').map(Number);
  let total = n1+n2;

  if (total%2 == 0) {
    if (e1 == 'PAR') {
      console.log(nome1);
    } else {
      console.log(nome2);
    }
  } else {
    if (e1 == 'IMPAR') {
      console.log(nome1);
    } else {
      console.log(nome2);
    }
  } 
}
